support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -261,8 +276,10 @@ function assignUserToVariant(
 		}
 
 		return index;
-	} else if (referrerControlled && acquisitionDataTest === null) {
+	} else if (referrerControlled && acquisitionDataTests === undefined) {
 		console.error('A/B test expects acquistion data but none was provided');
+
+		return -1;
```

Comment:
Before, we weren't returning from this branch. This means we would fall through to line 285 and return a random variant. I think this probably isn't the best behaviour. The test has been set up as `referrerControlled` but we're assigning them randomly. I think if it's `referrerControlled` but we can't find the assignment in `acquisitionData` we shouldn't assign them to that test

Reply:
This makes sense.
Side note - your change didn't introduce this, but giving `-1` a special meaning always feels a little unnatural/brittle. I wonder if it could return `number | notintest` instead? Not something that needs addressing now!

Reply:
Yeah I'm having a go at some more refactors now to clean this file up a bit, but will do that as a follow up PR

Reply:
Remvoing the `-1` is definitely on the todo list!